x-pack/test/api_integration/apis/ml/setup_module.ts

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+import { JOB_STATE, DATAFEED_STATE } from '../../../../plugins/ml/common/constants/states';
+import { USER } from '../../../functional/services/machine_learning/security_common';
+
+const COMMON_HEADERS = {
+  'kbn-xsrf': 'some-xsrf-token',
+};
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const testDataListPositive = [
+    {
+      testTitleSuffix: 'for sample logs dataset with prefix and startDatafeed false',
+      sourceDataArchive: 'ml/sample_logs',
+      module: 'sample_data_weblogs',
+      user: USER.ML_POWERUSER,
+      requestBody: {
+        prefix: 'pf1_',
+        indexPatternName: 'kibana_sample_data_logs',
```

Comment:
nit, `indexPatternName` isn't needed here as it's the same as the module's default index pattern name.
A future test could compare the results of calling `setup` with and without `indexPatternName`

Reply:
Thanks for the hint @jgowdyelastic, I'll add more tests for the request parameters in the future, including the `setup` call with and without `indexPatternName`.